hw/formal/tools/jaspergold/parse-fpv-report.py

```diff
@@ -0,0 +1,201 @@
+            coverage_patterns = [("stimuli", r"^\|\w+.*\|(\d*.\d\d)%.*\|\d*.\d\d%.*\|\d*.\d\d%.*"),
+                                 ("coi", r"^\|\w+.*\|\d*.\d\d%.*\|(\d*.\d\d)%.*\|\d*.\d\d%.*"),
+                                 ("proof", r"^\|\w+.*\|\d*.\d\d%.*\|\d*.\d\d%.*\|(\d*.\d\d)%.*")]
+            cov_results.update(extract_messages(full_file, coverage_patterns))
+            for key, item in cov_results.items():
+                if len(item) == 1:
+                    cov_results[key] = item[0] + " %"
+                else:
+                    cov_results[key] = "N/A"
+                    log.error("Parse %s coverage error. Expect one matching value, get %s",
+                              key, item)
+
+    except IOError as err:
+        log.error("[get_cov_results] IOError %s", err)
+        cov_results = None
+
+    return cov_results
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=
+        '''This script parses the JasperGold FPV log to extract below information.
+
+        "messages": {
+          "errors"      : []
+          "warnings"    : []
+          "cex"         : ["property1", "property2"...],
+          "undetermined": [],
+          "unreachable" : [],
+        },
+        "fpv_summary": {
+          "errors"      : 0
+          "warnings"    : 2
+          "proven"      : 20,
+          "cex"         : 5,
+          "covered"     : 18,
+          "undetermined": 7,
+          "unreachable" : 2,
+          "pass_rate"   : "90 %",
+          "cover_rate"  : "90 %"
+        },
+        If coverage is enabled, this script will also parse the coverage result:
+        "fpv_coverage": {
+          stimuli: "90 %",
+          coi    : "90 %",
+          proof  : "80 %"
+        }
+        ''')
+    parser.add_argument('--logpath',
+                        type=str,
+                        default="fpv.log",
+                        help=('FPV log file path. Defaults to `fpv.log` '
+                              'under the current script directory.'))
+
+    parser.add_argument('--reppath',
+                        type=str,
+                        default="results.hjson",
+                        help=('Parsed output hjson file path. Defaults to '
+                              '`results.hjson` under the current script directory.'))
+
+    parser.add_argument('--cov',
+                        type=int,
+                        default=0,
+                        help=('Enable parsing coverage data. '
+                              'Default coverage parsing is disabled.'))
+
+    args = parser.parse_args()
+
+    results = get_results(args.logpath)
+
+    if args.cov:
+        results["fpv_coverage"] = get_cov_results(args.logpath)
+
+    with Path(args.reppath).open("w") as results_file:
+        hjson.dump(results,
+                   results_file,
+                   ensure_ascii=False,
+                   for_json=True,
+                   use_decimal=True)
+
```

Comment:
since dvsim looks at the status returned by this script to determine whether FPV was successful or not, should we do something similar to the lint case and return with 0 only if there where no errors/warnings, and all properties have been proven?